libraries/AP_HAL/Util.h

```diff
@@ -34,6 +34,16 @@ class AP_HAL::Util {
     virtual const char* get_custom_terrain_directory() const { return nullptr;  }
     virtual const char *get_custom_storage_directory() const { return nullptr;  }
 
+    // set number of available cores to use
+    virtual void set_affinity_core_num (int affinity_core_num) {};
+    virtual int get_affinity_core_num () const { return 0;  }
+    // set core index of the first core we can use. e.g. Isolated cores should start from 1
+    virtual void set_affinity_core_start_index (int affinity_core_start_index) {};
+    virtual int get_affinity_core_start_index () const { return 0;  }
+    // set current core to use
+    virtual void set_affinity_core_cur (int affinity_core_cur) {};
+    virtual int get_affinity_core_cur ()  const { return 0;  }
+
     // get path to custom defaults file for AP_Param
```

Comment:
do we really need this? Is it useful for other non-Linux boards?

Reply:
I thought we should base class consistent and these cpu affinity could be implemented later when required.
anyway, I have removed them.

Reply:
please bare with me.

I set RPI  **isolcpus=1,2** so now the any program will launch in cpu 0 & 3 only.

Now I use the command:

`pi@raspberrypi:~ $ sudo /home/pi/arducopter -A udp:192.168.1.144:14550:bcast -B /dev/serial0 -C /dev/ardu --cpu-affinity 6`

This command it instructs ardupilot to use cpu mask 0b0110 which means that core 1 & core 2 should be used. 
What I expect from this command is to run all threads on cpu 1 & 2 and that what happens.

pi@raspberrypi:~ $ ps -mo pid,tid,%cpu,psr -C arducopter
  PID   TID %CPU PSR
 1364     - 10.6   -
    -  1364  4.6   1
    -  1365  1.6   2
    -  1366  0.3   1
    -  1367  2.3   2
    -  1368  0.0   1
    -  1369  0.5   2
    -  1370  1.5   1

Yes at the beginner I should move the main thread to cpu 1 or cpu 2 as it will be launched by OS on cpu 0 & 3 due to **isolcpus** values.

but I dont all threads to inherit the same cpui. I want to be able to put them on different cpus.